### PR TITLE
feat(DTFS2-7230): added yaml validation

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction :pencil2:
Ensure all `*.yml` files are validated in the pipeline.

## Resolution :heavy_check_mark:
* Added `yaml-lint` with GLOB path syntax as a NPM script `validate:yml` and to staged files using `lint-staged` script.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -130,3 +129,31 @@ jobs:
...
+    steps:
+      - name: Timezone
+        uses: szenius/set-timezone@v1.2
+        with:
+          timezoneLinux: ${{ needs.setup.outputs.timezone }}
+
+      - name: Repository
+        uses: actions/checkout@v4
+
+      - name: Node
```
Comment (AlexBramhill) [inline .github/workflows/lint.yml:148]: We probably don't need to npm i here as we're using `npx`? to run yaml-lint?  Maybe we could have a locally runable npm script for this for devs to use?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -130,3 +129,31 @@ jobs:
...
         run: npm run prettier:all
+
+  # 4. YAML
+  yaml:
+    name: YAML 🎨
+    environment:
+      name: ${{ needs.setup.outputs.environment }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Timezone
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply .github/workflows/lint.yml:140]: For error logging to ensure it is in UTC.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (abhi-markan) [reply .github/workflows/lint.yml:148]: Good shout, have added as `validate:yml` script.

File: package.json [hunk 3]
```diff
@@ -46,13 +46,15 @@
     "start:detach": "npm run start -- --detach",
     "stop": "docker compose down",
     "type-check:all": "npm run type-check --workspaces --if-present",
-    "update:all": "npm update --save --if-present"
+    "update:all": "npm update --save --if-present",
+    "validate:yml": "npx yaml-lint **/*.yml"
```
Comment (AlexBramhill) [inline package.json:50]: `lint:yml`?

Comment (abhi-markan) [reply package.json:50]: This script is validating the YML syntax only and not linting it.

**Review by AlexBramhill — APPROVED** [review]

**Pull request merged**